src/Input/TextField/TextField.test.tsx

```diff
@@ -0,0 +1,148 @@
+import * as React from 'react';
+
+import * as renderer from 'react-test-renderer';
+import '@testing-library/jest-dom/extend-expect';
+import { render } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+
+import TextField, { textFieldType } from './TextField';
+import { SecondaryColor, PrimaryColor } from '../../Style/Colors';
+
+const props = {
+  type: 'text' as textFieldType,
+  label: 'Username',
+  onChange: jest.fn().mockImplementation(event => event.target.value),
+};
+
+const inputValue = 'glintsaries';
+
+function setupTextField(otherProps?: any) {
+  const { getByLabelText, ...utils } = render(
+    <TextField
+      type={props.type}
+      label={props.label}
+      onChange={props.onChange}
+      {...otherProps}
+    />
+  );
+  const textFieldInput = getByLabelText(props.label) as HTMLInputElement;
+  return { textFieldInput, utils };
+}
+
+function setupTextFieldWithChangeEvent(otherProps?: any) {
+  const { textFieldInput } = setupTextField(otherProps);
+  userEvent.type(textFieldInput, inputValue);
+  return { textFieldInput };
+}
+
+it('<TextField> should render with a label of Username and type of text', () => {
+  const TextFieldSnapshot = renderer
+    .create(<TextField type={props.type} label={props.label} />)
+    .toJSON();
+  expect(TextFieldSnapshot).toMatchSnapshot();
+});
+
+describe('when it is rendered', () => {
+  it('should display the correct input label', () => {
+    const { textFieldInput } = setupTextField();
+    expect(textFieldInput).toBeTruthy();
```

Comment:
should we check the label value?

Reply:
This line in `setupTextField` already checks for it.

```
const textFieldInput = getByLabelText(props.label) as HTMLInputElement;
```

Reply:
oh yeah